scripts/create-changelog-pr.js

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env node
+
+const semver = require('semver')
+const { compileChangeLog, compileReleaseNotesMd, removeEntries, addReleaseNotesToChangelogMD } = require("./libs/changelog")
+const path = require('path');
+const { execSync } = require('child_process');
+const fs = require('fs');
+
+let BUILD_DIR = path.join(".", "build")
+const CHANGELOG_PATH = path.join(".", "CHANGELOG.md")
+
+try {
+    main()
+} catch (err) {
+    console.error(err)
+    console.error('Usage: scripts/changelog-pr --version [release version, for example 1.0.0] [--branch main] [--dry-run]');
+}
+
+function main() {
+    let args = parseArguments()
+    // TODO: check if GH exists
+    // TODO: check if branch matches current one
+    console.log("Generating dependencies.md")
+    execSync("./gradlew createDependenciesMd")
+    let dependenciesMd = fs.readFileSync(path.join(".", "build", "dependencies.md"))
+    console.log("Generated dependencies.md")
+
+    console.log("Compiling changelog")
+    let changelog = compileReleaseNotesMd({
+        version: args.version,
+        dependenciesMd: dependenciesMd
+    })
+
+    let executor = args.isDryRun
+        ? execSyncDryRun
+        : execSync
+
+    let releaseNotesTempFile = path.join(BUILD_DIR, "RELEASENOTES.md")
+    if (args.isDryRun) {
+        console.log("Dry run. Generated changelog:")
+        console.log(changelog)
+    } else {
+        fs.writeFileSync(releaseNotesTempFile, changelog)
+        removeEntries()
+        updateChangelogMDFile(changelog)
+    }
+
+
+    executor(`git checkout -b add-changelog-${args.version}`)
+    executor(`git add .`)
+    executor(`git commit -m "created changelog for ${args.version}"`)
+    executor(`gh config set prompt disabled`)
+    executor(`git push --set-upstream origin add-changelog-${args.version}`)
+    executor(`gh pr create --base ${args.branch} --title "Changelog for ${args.version}" --body "" --reviewer mapbox/navigation-android`)
+    executor(`gh release create ${args.version} --draft --target ${args.branch} --notes-file ${releaseNotesTempFile} --title ${args.version}`)
+    if (args.branch != "main") {
+        executor(`git checkout main`)
+        executor(`git checkout -b add-changelog-${args.version}-update-main`)
+        if (args.isDryRun) {
+            console.log("updating changelog at main branch")
+        } else {
+            updateChangelogMDFile(changelog)
+        }
+        executor(`git add .`)
+        executor(`git commit -m "Changelog for ${args.version}"`)
+        executor(`git push origin add-changelog-${args.version}-update-main`)
+        executor(`gh pr create --base main --title "Changelog for ${args.version}" --body "" --reviewer mapbox/navigation-android`)
+    }
+    executor(`git checkout ${args.branch}`)
```

Comment:
If somebody launches the script from a not main branch, for example `release-2.1`, script opens 2 PRs with updates in the `CHANGELOG.md` file, one for the `release-2.1` and one for the `main` branch.
@mapbox/navigation-android , @LukasPaczos , do you like this approach?

Alternatively we can update `CHANGELOG.md` automatically in the main branch only and update a release branch manually or not to update a release branch at all.

Reply:
Hm. Not I'm not sure that this is an approach we use. All our changes go through the master branch and then they're cherry-picked to release branch. 🤔 
Guys I need help with this 😄